### DISCUSSION TITLE
Interesting unexpected behavior with nicegui/examples/authentication/main.py in current Firefox

### DISCUSSION THREAD

**Discussion by vmlstandard** [zauberzeug/nicegui#3530] [Q&A]
### Question

- Run this file: nicegui/examples/authentication/main.py
- Type: user1 / pass1 -> it logs-in with no problem, 
- Firefox offers to save login / password -> agree
- Log Out
- Firefox auto populates login and password, since it was saved
- Click on Log-In -> it fails to login

Could be Firefox bug

What is weird, if you also save: user2 / pass2 in Firefox, then both logins / passwords work as expected with no errors

**Comment by rodja** [comment]
It works for me. Can anyone else reproduce it?

**Reply by vmlstandard** [reply]
I have tested this scenario on Mac (Sonoma) and now, on Mint PC - the same issue.

Additional info, if it matters: I'm using PyCharm CE 2024.1.3 to run this Auth Demo, but, I'm pretty sure, that shouldn't matter.

Screenshots of the steps to reproduce:
[Issue with Saving and reusing credentials (Firefox).pdf](https://github.com/user-attachments/files/16646276/Issue.with.Saving.and.reusing.credentials.Firefox.pdf)

Same thing, if I run the **nicegui/examples/authentication/main.py** from command line

python -V
Python 3.12.4

**Reply by rodja** [reply]
Oh wow. I've must done something wrong earlier. Now I can reproduce the issue. Sorry.

**Comment by rodja** [comment]
It looks like the input fields are filled by Firefox before we have a client connection. Because we do not buffer client-side changes and transmit them on connection, Python is not aware of the filled values. If you have more than one saved password, Firefox will ask you to chose. And then, the client connection is established. There are two possible fixes: 1. send all input values on client connection (NiceGUI core) or add `await ui.context.client.connected()` at the beginning of the `login()` function in the demo (user code). What do you think @falkoschindler?

**Reply by falkoschindler** [reply]
The client could wait for the handshake before sending events to the server. I'll think about it and create an issue and/or PR.

**Reply by falkoschindler** [reply]
Minimum reproduction:
```py
username = ui.input('Username')
password = ui.input('Password', password=True, password_toggle_button=True)
ui.button('Login', on_click=lambda: ui.notify(f'Logging in as {username.value} with password {password.value}'))
```
Saving the credentials with Firefox and re-loading the page auto-fills the inputs without sending the input to the server, so the notification shows and empty username and password.

**Reply by daya0576** [reply]
Thanks for sharing the root cause and saving my day XD

I ran into the same issue with Bitwarden autofill and couldn’t reproduce it locally. The solution worked for me, too.

**Comment by vmlstandard** [comment]
Thank you, rodja,

The second fix worked for me.

BTW, only case if only one password saved in Firefox for the Test program's url is causing the problem above.

If two different users / passwords are saved - both work fine without any fixes.

But now the problem is solved:

I have changed:

```py
@ui.page('/login')
async def login() -> Optional[RedirectResponse]:
    await ui.context.client.connected()
```

to:

```py
@ui.page('/login')
async def login() -> Optional[RedirectResponse]:
    await ui.context.client.connected()
```

and it works as it should - saved user logs-in


the future maintenance core fix sounds reasonable
